openshift/openshift/src/main/java/org/arquillian/cube/openshift/impl/graphene/location/OpenshiftCustomizableURLResourceProvider.java

```diff
@@ -0,0 +1,97 @@
+package org.arquillian.cube.openshift.impl.graphene.location;
+
+import io.fabric8.openshift.api.model.v3_1.Route;
+import org.arquillian.cube.openshift.impl.client.OpenShiftClient;
+import org.arquillian.cube.kubernetes.api.Configuration;
+import org.arquillian.cube.openshift.impl.client.CubeOpenShiftConfiguration;
+import org.jboss.arquillian.core.api.Instance;
+import org.jboss.arquillian.core.api.annotation.Inject;
+import org.jboss.arquillian.graphene.spi.configuration.GrapheneConfiguration;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.arquillian.test.spi.enricher.resource.ResourceProvider;
+
+import java.lang.annotation.Annotation;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Optional;
+
+public class OpenshiftCustomizableURLResourceProvider implements ResourceProvider {
+
+    @Inject
+    private Instance<GrapheneConfiguration> grapheneConfiguration;
+
+    @Inject
+    private Instance<OpenShiftClient> clientInstance;
+
+    @Inject
+    private Instance<Configuration> configurationInstance;
+
+    @Override
+    public boolean canProvide(Class<?> type) {
+        return URL.class.isAssignableFrom(type);
+    }
+
+    @Override
+    public Object lookup(ArquillianResource resource, Annotation... qualifiers) {
+        try {
+            return resolveUrl();
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    private URL resolveUrl() throws MalformedURLException {
+        final GrapheneConfiguration grapheneConfiguration = this.grapheneConfiguration.get();
+
+        final String configuredUrl = grapheneConfiguration.getUrl();
+
+        if (configuredUrl != null && !configuredUrl.isEmpty()) {
+            return getRoute(new URL(configuredUrl).getHost()).get();
+        }
+        return getRoute().get();
+    }
+
+    private Optional<URL> getRoute(String routeName) {
```

Comment:
Instead of passing routeName, pass the URL (`new URL(configuredURL)`) then do the `getHost` thing. But if `getRoute` returns null, return the original `new URL(configuredURL)`. So in this way this method does not return an `Optional` and always returns a valid `URL` object.